src/components/Section/ProposalUpdatesActions.tsx

```diff
@@ -14,17 +20,40 @@ import './ProposalUpdatesActions.css'
 type ProposalUpdatesActionsProps = {
   nextUpdate?: UpdateAttributes | null
   currentUpdate?: UpdateAttributes | null
-  onPostUpdateClick: () => void
+  pendingUpdates?: UpdateAttributes[]
+  proposal: ProposalAttributes | null
 }
 
 export default function ProposalUpdatesActions({
-  onPostUpdateClick,
   nextUpdate,
   currentUpdate,
+  pendingUpdates,
+  proposal,
 }: ProposalUpdatesActionsProps) {
   const t = useFormatMessage()
   const hasSubmittedUpdate = !!currentUpdate?.completion_date
 
+  const { isOwner } = useIsProposalOwner(proposal)
+  const { isCoauthor } = useIsProposalCoAuthor(proposal)
+  const showProposalUpdatesActions =
+    isProposalStatusWithUpdates(proposal?.status) && proposal?.type === ProposalType.Grant && (isOwner || isCoauthor)
+
+  const onPostUpdateClick = useCallback(() => {
+    if (proposal === null) {
```

Comment:
If render condition is moved to parent component, we can remove null as part of the proposal type and remove this check.